libflux/src/flux/semantic/types.rs

```diff
@@ -47,9 +46,20 @@ impl PartialEq for PolyType {
         let mut f = Fresher::from(max + 1);
         let mut g = Fresher::from(max + 1);
 
-        self.clone()
-            .fresh(&mut f)
-            .equal(&poly.clone().fresh(&mut g))
+        let mut a = self.clone().fresh(&mut f, &mut HashMap::new());
+        let mut b = poly.clone().fresh(&mut g, &mut HashMap::new());
+
+        a.vars.sort();
```

Comment:
I wonder if there's a way to ensure that we have a deterministic, canonical ordering of type variables within a polytype as soon as its created, so we don't need to do any sorting later on when we need to compare.  Curious about your thoughts on this.

Reply:
We could use a btree to ensure that. This code will only ever be invoked during tests though. Is your concern about performance or regressions?

Reply:
My concern was mostly performance.  Seeing an `eq` method that calls sort multiple times gives me pause.  If it's only used in tests, then I'm not that concerned.